subsys/bluetooth/controller/ll_sw/ull_sync_iso.c

```diff
@@ -11,29 +11,293 @@
 #include "common/log.h"
 #include "hal/debug.h"
 
+#include <sys/byteorder.h>
+
+#include "util/util.h"
+#include "util/mem.h"
+#include "util/memq.h"
+#include "util/mayfly.h"
+
+#include "hal/ccm.h"
+#include "hal/radio.h"
+#include "hal/ticker.h"
+
+#include "ticker/ticker.h"
+
+#include "pdu.h"
+#include "ll.h"
+
+#include "lll.h"
+#include "lll_vendor.h"
+#include "lll_clock.h"
+#include "lll_scan.h"
+#include "lll_sync.h"
+#include "lll_sync_iso.h"
+
+#include "ull_scan_types.h"
+#include "ull_sync_types.h"
+
+#include "ull_internal.h"
+#include "ull_scan_internal.h"
+#include "ull_sync_internal.h"
+#include "ull_sync_iso_internal.h"
+
+static struct ll_sync_iso ll_sync_iso_pool[CONFIG_BT_CTLR_SCAN_SYNC_ISO_MAX];
+static void *sync_iso_free;
+
+static int init_reset(void);
+static inline struct ll_sync_iso *sync_iso_acquire(void);
+static void ticker_cb(uint32_t ticks_at_expire, uint32_t remainder,
+		      uint16_t lazy, void *param);
+static void ticker_op_cb(uint32_t status, void *param);
+
 uint8_t ll_big_sync_create(uint8_t big_handle, uint16_t sync_handle,
 			   uint8_t encryption, uint8_t *bcode, uint8_t mse,
 			   uint16_t sync_timeout, uint8_t num_bis,
 			   uint8_t *bis)
 {
-	/* TODO: Implement */
-	ARG_UNUSED(big_handle);
-	ARG_UNUSED(sync_handle);
-	ARG_UNUSED(encryption);
-	ARG_UNUSED(bcode);
-	ARG_UNUSED(mse);
-	ARG_UNUSED(sync_timeout);
-	ARG_UNUSED(num_bis);
-	ARG_UNUSED(bis);
+	memq_link_t *big_sync_estab;
+	memq_link_t *big_sync_lost;
+	struct lll_sync_iso *lll_sync;
+	struct ll_sync_set *sync;
+	struct ll_sync_iso *sync_iso;
 
-	return BT_HCI_ERR_CMD_DISALLOWED;
-}
+	sync = ull_sync_is_enabled_get(sync_handle);
+	if (!sync || sync->sync_iso) {
+		return BT_HCI_ERR_CMD_DISALLOWED;
+	}
+
+	sync_iso = sync_iso_acquire();
+	if (!sync_iso) {
+		return BT_HCI_ERR_MEM_CAPACITY_EXCEEDED;
+	}
+
+	big_sync_estab = ll_rx_link_alloc();
+	if (!big_sync_estab) {
+		return BT_HCI_ERR_MEM_CAPACITY_EXCEEDED;
+	}
+
+	big_sync_lost = ll_rx_link_alloc();
+	if (!big_sync_lost) {
+		ll_rx_link_release(big_sync_estab);
+
+		return BT_HCI_ERR_MEM_CAPACITY_EXCEEDED;
+	}
 
+	sync_iso->node_rx_lost.link = big_sync_lost;
+	sync_iso->node_rx_estab.link = big_sync_estab;
+
+	lll_sync = &sync_iso->lll;
+
+	/* Initialise ULL and LLL headers */
+	ull_hdr_init(&sync->ull);
```

Comment:
Should this be `ull_hdr_init(&sync_iso->ull)`?

Reply:
Yes! Great catch